kratos/solving_strategies/convergencecriterias/displacement_criteria.h

```diff
@@ -436,7 +441,7 @@ class DisplacementCriteria
         struct TLS {TDataType dof_value{};};
 
         const TDataType reference_disp_norm = block_for_each<SumReduction<TDataType>>(rDofSet, TLS(), [this, &rank](auto& rDof, TLS& rTLS) {
```

Comment:
why don't you check for distributed or not distributed outside of this loop? this would allow avoiding the if constexpr in the IsFreeDof

also, you did not answer to my comment about the why the partition index check is needed

and indeed if we can do this without any dependency on trilinos, than i would only have the base class and instantiate it with different template arguments from within the trilinos_applicaiton as suggested by @roigcarlo

Reply:
I have a better solution in mind. My point is to avoid the penalty of that check and minimize the code duplication

Reply:
Fixed with https://github.com/KratosMultiphysics/Kratos/pull/11846, now I defined a static method in the spaces that provide information if distributed or not. If a new space is considered a compilation error will appear and the method is implemented in less than 1 minute, I think is a good solution. This avoid any penalty in performance as it is a static check.